lib/directory-tree.js

```diff
@@ -1,54 +1,65 @@
 var fs = require('fs');
 var path = require('path');
 
-function directoryTree(basepath, extensions) {
-    var _directoryTree = function (name, extensions) {
-        /*
-         * Certain symbolic links/weird files break when you try to stat them
-         */
-        try {
-            var stats = fs.statSync(name);
-        } catch(e) {
-            return null;
+function directoryTree(basepath, myOptions) {
+    var _directoryTree = function (name, myOptions) {
+        
+        if(!myOptions){
+            myOptions = [];
         }
+        
+        var options = {
+            'hideFiles': myOptions.hideFiles || false,
+            'hideEmptyDirectories': myOptions.hideEmptyDirectories ||false,
+            'ignoreList': myOptions.ignoreList || [],
+            'fileExtensions': myOptions.fileExtensions || []
+        };
+
+        var stats = fs.statSync(name);
 
         var item = {
             path: path.relative(basepath, name),
             name: path.basename(name)
         };
 
+        if (stats.isFile() && options.hideFiles) {
+            return null;
+        }
+
+        if (options.ignoreList.length > 0) { 
+            var ignore = false;
+            options.ignoreList.forEach(function (value) {
+                if (name.indexOf(value) > -1) { 
+                    ignore = true;
+                    return null;
```

Comment:
You don't really need this return statement, forEach will execute on every element anyway.